contracts/LibSubmarineSimple.sol

```diff
@@ -0,0 +1,171 @@
+pragma solidity ^0.4.24;
+
+import "./SafeMath.sol";
+import "./proveth/ProvethVerifier.sol";
+import "./proveth/RLP.sol"; // probably can remove this line, since ProvethVerifier also imports RLP.sol
+
+contract LibSubmarineSimple is ProvethVerifier {
+
+    using SafeMath for uint256;
+
+    ////////////
+    // Events //
+    ////////////
+
+    event Unlocked(
+        bytes32 indexed _commitId,
+        uint96 _commitValue
+    );
+    event Revealed(
+        bytes32 indexed _commitId,
+        uint96 _commitValue,
+        bytes32 _witness,
+        bytes32 _commitBlockHash,
+        address submarineAddr
+    );
+
+    /////////////
+    // Storage //
+    /////////////
+
+    uint8 public vee = 27; // the ECDSA v parameter: 27 allows us to be broadcast on any network (i.e. mainnet, ropsten, rinkeby etc.)
+    uint8 public commitPeriodLength = 20; // How many blocks must a submarine be committed for before being revealed - 20 by default, but API clients are free to modify this
+
+    mapping(bytes32 => CommitData) public commitData; // stored "session" state information
+
+    // A submarine send is considered "finished" when the amount revealed and unlocked are both greater than zero, and the amount for the unlock is greater than or equal to the reveal amount.
+    struct CommitData {
+        uint96 amountRevealed; // amount the reveal transaction revealed would be sent in wei. When greater than zero, the submarine has been revealed.
+        uint96 amountUnlocked; // amount the unlock transaction recieved in wei. When greater than zero, the submarine has been unlocked; however the submarine may not be finished, until the unlock amount is GREATER than the promised revealed amount.
+        uint32 blockNumber; // what block was this revealed in?
```

Comment:
Should be the block in which it was committed. Not the block in which it was revealed.

Reply:
Fixed

Reply:
derp
I'm so glad we have multiple eyes on this code